### DISCUSSION TITLE
Should budget/s respect unpaid bills?

### DISCUSSION THREAD

**Discussion by qck4fun** [firefly-iii/firefly-iii#7936] [General]
I'm having a budget and I always check how much money I have left with for the day, but this amount apparently doesn't count in unpaid bills, which I also have for the period.

The dilemma is I know the bills are going to be paid anyway, so why they're not included in the calculation. Sure I can do the calculation myself, but I'm wondering whether this is intentional or not.

Thanks for the feedback :pray:

**Comment by JC5** [comment]
Firefly III will not pre-fill your budgets based on bills, this is intentional. There is also no link between bills and budgets. I'm happy to check out the results of the poll (I shall not vote myself) but this behavior will not be changed I'm afraid.

**Reply by qck4fun** [reply]
Well, I guess I'll be the only one voting here :smile:

**Reply by EricVanCaenenberghe** [reply]
Just adding a struggle I have between Contracts and Budgets. For all my recurring bills I have a contract. In the dashboard it has a dedicated forecast: left to pay and already paid so I don't create a budget for those foreseen budgets. Contracts are planned vs Budget is more unplanned (okay you need groceries and you eat out but it is not contract based) . Works fine by me on the dashboard.

But in the pie-charts in the expenses or in the overview reports however those contracts pollute my budget view: there is pie indicating not-budgetted. In fact it was budgetted but through the contracts.

One reason of not budgetting the outstanding contracts is that I have to calculate each month (my dashboard period) the budget manually based on the contracts (while dashboard I have this little overview).

My first idea would be: why not generate a contract based budget then the not-budgetted pie is really not budgetted ... but then in the top of the dashboard contracts are doubled: once in the contract section, once in the budget section ...
Another possibility: the not-budgetted pie does not include the contract based not-budgetted items (have fun)

AND there should be a pie-chart in the expenses and reports about the contracts.

Meanwhile still in doubt about the vote.

**Comment by devfaz** [comment]
Workaround: define 2 budgets. One for known bills, one for "random" stuff. This way you know how much you can spend on random stuff.

**Reply by qck4fun** [reply]
Mmm! Sounds like a neat workaround! :1st_place_medal: thank you! I'll try it
